src/components/organisms/SearchPane/SearchPane.tsx

```diff
@@ -0,0 +1,524 @@
+    }
+    if (queryMatchParam.regExp) {
+      queryRegExp = new RegExp(query, matchParams);
+    }
+
+    const filteredFileMap: FileEntry[] = Object.values(fileMap)
+      .map(file => filterFileMap(file, queryRegExp))
+      .filter(v => Boolean(v));
+
+    const treeData = createFilteredNode(filteredFileMap);
+
+    setSearchTree(treeData);
+    saveQueryHistory(query);
+  };
+
+  const handleSearchQueryChange = (e: {target: HTMLInputElement}) => {
+    updateSearchQuery(e.target.value);
+
+    debounceHandler.current && clearTimeout(debounceHandler.current);
+    debounceHandler.current = setTimeout(() => {
+      findMatches(e.target.value);
+    }, 1000);
+  };
+
+  const toggleMatchParam = (param: keyof MatchParamProps) => {
+    setQueryMatchParam(prevState => ({...prevState, [param]: !prevState[param]}));
+  };
+
+  return (
+    <S.FileTreeContainer id="AdvancedSearch">
+      <TitleBar title="Advanced Search" closable />
+      <S.TreeContainer>
+        <S.SearchBox>
+          <Input placeholder="Search anything..." value={searchQuery} onChange={handleSearchQueryChange} />
+          <S.StyledButton $isItemSelected={queryMatchParam.matchCase} onClick={() => toggleMatchParam('matchCase')}>
+            Aa
+          </S.StyledButton>
+          <S.StyledButton
+            $isItemSelected={queryMatchParam.matchWholeWord}
+            onClick={() => toggleMatchParam('matchWholeWord')}
+          >
+            [
+          </S.StyledButton>
+          <S.StyledButton $isItemSelected={queryMatchParam.regExp} onClick={() => toggleMatchParam('regExp')}>
+            .*
+          </S.StyledButton>
+        </S.SearchBox>
+        <S.RootFolderText>
+          {searchTree ? (
+            <S.MatchText id="file-explorer-search-count">
+              {searchCounter.current.totalMatchCount} matches in {searchCounter.current.filesCount} files
+            </S.MatchText>
+          ) : (
+            <RecentSearch
+              handleClick={query => {
+                updateSearchQuery(query);
+                findMatches(query);
+              }}
+            />
+          )}
+        </S.RootFolderText>
+
+        {searchTree && (
+          <S.TreeDirectoryTree
+            height={height - 2 * DEFAULT_PANE_TITLE_HEIGHT - 20}
+            onSelect={onSelect}
+            treeData={[searchTree]}
+            ref={treeRef}
+            expandedKeys={expandedFolders}
+            onExpand={onExpand}
+            titleRender={event => (
+              <TreeItem
+                treeKey={String(event.key)}
+                title={event.title}
+                processingEntity={processingEntity}
+                setProcessingEntity={setProcessingEntity}
+                onDelete={onDelete}
+                onRename={onRename}
+                onExcludeFromProcessing={onExcludeFromProcessing}
+                onIncludeToProcessing={onIncludeToProcessing}
+                onCreateFolder={onCreateFolder}
```

Comment:
Probably need to separate this functionality of all these onCreateFolder, onDelete, onPreview etc methods and reuse it both in FilterTree and SearchPane so not to duplicate all these methods in both components